plugins/module_utils/prerequisites.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+
+# (c) 2023, Jan Kodera (jkodera@purestorage.com)
+# GNU General Public License v3.0+ (see COPYING.GPLv3 or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import sys
+import re
+import importlib
+import importlib.metadata
+import json
+
+# This file exists because Ansible currently cannot declare dependencies on Python modules
+# when imported. When imported, it fails with Ansible-compatible message if requirements
+# listed below are not installed
+# see https://github.com/ansible/ansible/issues/62733 for more info about lack of req support
+
+#############################
+
+# 'module_name, package_name, version_requirements' triplets
+DEPENDENCIES = [
+    ("fusion", "purefusion", ">=1.0.11,<2.0")
+]
+
+#############################
+
+# returns tuple (MAJOR, MINOR, PATCH)
+def _parse_version(val):
+    # regexes for this were really ugly
+    try:
```

Comment:
You can do parsing via official semver regex
```
semver_regex = re.compile(r'^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$')
```
match = semver_regex.match(semver_regex)
if not match:
    return None
return match.group('major'), match.group('minor), match.group('patch')
```